server/svix-server/src/error.rs

```diff
@@ -31,57 +28,71 @@ pub type Result<T> = std::result::Result<T, Error>;
 #[derive(Debug)]
 pub struct Error {
     // the file name and line number of the error. Used for debugging non Http errors
-    pub trace: Vec<&'static str>,
+    pub trace: Vec<&'static Location<'static>>,
     pub typ: ErrorType,
 }
 
 impl Error {
-    pub fn generic(s: impl fmt::Display, location: &'static str) -> Self {
+    #[track_caller]
+    pub fn generic(s: impl fmt::Display) -> Self {
         Self {
-            trace: Self::init_trace(location),
+            trace: vec![Location::caller()],
             typ: ErrorType::Generic(s.to_string()),
         }
     }
 
-    pub fn database(s: impl fmt::Display, location: &'static str) -> Self {
+    #[track_caller]
+    pub fn database(s: impl fmt::Display) -> Self {
         Self {
-            trace: Self::init_trace(location),
+            trace: vec![Location::caller()],
             typ: ErrorType::Database(s.to_string()),
         }
     }
 
-    pub fn queue(s: impl fmt::Display, location: &'static str) -> Self {
+    #[track_caller]
+    pub fn queue(s: impl fmt::Display) -> Self {
         Self {
-            trace: Self::init_trace(location),
+            trace: vec![Location::caller()],
             typ: ErrorType::Queue(s.to_string()),
         }
     }
 
-    pub fn validation(s: impl fmt::Display, location: &'static str) -> Self {
+    #[track_caller]
+    pub fn validation(s: impl fmt::Display) -> Self {
         Self {
-            trace: Self::init_trace(location),
+            trace: vec![Location::caller()],
             typ: ErrorType::Validation(s.to_string()),
         }
     }
 
+    #[track_caller]
     pub fn http(h: HttpError) -> Self {
         Self {
-            trace: vec![], // no debugging necessary
+            trace: Vec::with_capacity(0), // no debugging necessary
             typ: ErrorType::Http(h),
         }
     }
 
-    pub fn cache(s: impl fmt::Display, location: &'static str) -> Self {
+    #[track_caller]
+    pub fn cache(s: impl fmt::Display) -> Self {
         Self {
-            trace: Self::init_trace(location),
+            trace: vec![Location::caller()],
             typ: ErrorType::Cache(s.to_string()),
         }
     }
 
-    fn init_trace(location: &'static str) -> Vec<&'static str> {
-        let mut trace = Vec::with_capacity(10); // somewhat arbitrary capacity, but avoids reallocation when building an error trace later on
-        trace.push(location);
-        trace
+    #[track_caller]
```

Comment:
Sorry, I just realized this, but we don't need `#[inline(always)]` here as well?

Reply:
Nope, that's not necessary for `track_caller` to work